### PR TITLE
look for metrics on -emu interface if it exists

### PR DESCRIPTION
**Pull request by eriknordmark**
We might be using -emu interfaces in some cases. This code transparently handles that case.

### CODE REVIEW

File: pkg/pillar/cmd/zedagent/handlemetrics.go [hunk 1]
```diff
@@ -591,7 +591,7 @@ func PublishMetricsToZedCloud(ctx *zedagentContext, cpuMemoryStat [][]string,
 		ReportAppMetric.Memory.AvailPercentage = availableMemoryPercent
 
 		appInterfaceList := aiStatus.GetAppInterfaceList()
-		log.Debugf("ReportMetrics: domainName %s ifs %v\n",
+		log.Infof("XXX ReportMetrics: domainName %s ifs %v\n",
```
Comment (kalyan-nidumolu) [inline pkg/pillar/cmd/zedagent/handlemetrics.go]: Will we keep spewing this? If yes - is Info a good level to be?

File: pkg/pillar/cmd/zedagent/handlemetrics.go [hunk 2]
```diff
@@ -647,6 +647,7 @@ func PublishMetricsToZedCloud(ctx *zedagentContext, cpuMemoryStat [][]string,
 			ReportAppMetric.Network = append(ReportAppMetric.Network,
 				networkDetails)
 		}
+		log.Infof("XXX ReportMetrics: network %+v", ReportAppMetric.Network)
```
Comment (kalyan-nidumolu) [inline pkg/pillar/cmd/zedagent/handlemetrics.go]: Same here.. Will we keep spewing this everytime we send metrics? Not very useful.. right? Retain it as debug?

File: pkg/pillar/cmd/zedmanager/updatestatus.go [hunk 3]
```diff
@@ -1032,6 +1042,43 @@ func doActivate(ctx *zedmanagerContext, uuidStr string,
...
+			changed = true
+		}
+	}
+
+	// for _, ec := range config.OverlayNetworkList {
+	return changed
+}
+
+// Lookup based on the name aka Vif
+func lookupVifInfo(ds types.DomainStatus, vif string) *types.VifInfo {
```
Comment (kalyan-nidumolu) [inline pkg/pillar/cmd/zedmanager/updatestatus.go]: This sounds like a good function to be a method of domainSTatus

ds.VifInfoByName(vif)

File: pkg/pillar/cmd/zedagent/handlemetrics.go (hunk 1 above)
Comment (eriknordmark) [reply pkg/pillar/cmd/zedagent/handlemetrics.go]: I'll fix those before merging. Thanks for reminding me.

File: pkg/pillar/cmd/zedagent/handlemetrics.go (hunk 2 above)
Comment (eriknordmark) [reply pkg/pillar/cmd/zedagent/handlemetrics.go]: Removed

**Comment by eriknordmark** [comment]
> What is the end-2-end flow here? How is this configured in the controller? trying to get a bigger picture of what we are doing here.

@kalyan-nidumolu 
We are likely to use type=ioemu for the vifs in some cases (those details are being worked separately) as opposed to the type=vif we use now. When we have a vif with type=ioemu then Xen creates a separate ip link with a "-emu" appended it to the name and attaches it to the same bridge, and that -emu is used for all of the traffic hence the UP/NO-CARRIER and interface counters are on the -emu interface.

Thus when the xen config file creates a vif named "nbu1x1" that IP link gets created plus a "nbu1x1-emu". 

This PR makes the device check if the -emu exists, and it is does it uses that to report status and metrics. Otherwise it uses the name of the Vif.

File: pkg/pillar/cmd/zedmanager/updatestatus.go (hunk 3 above)
Comment (eriknordmark) [reply pkg/pillar/cmd/zedmanager/updatestatus.go]: Since the "name" is in the Vif field I'll name it ds.VifInfoByVif()

**Comment by kalyan-nidumolu** [comment]
Ah ok.. Thanks for the explanation - So this is nothing related to audio. These are network interfaces, but *some* yet to be decided config will cause us to use the -emu version.. 

I am a bit concerned about checking presence of Linux devices and deciding whether to look at <dev> or <dev>-emu. Shouldn't this info be coming from some config?

**Comment by eriknordmark** [comment]
> I am a bit concerned about checking presence of Linux devices and deciding whether to look at or -emu. Shouldn't this info be coming from some config?

Well, ideally Xen shouldn't be adding an additional ip link so I get the name I expect, so we're far from ideal. The code which determines whether we use type=ioemu or something else will depend on the config, but the dependency isn't clear. Dynamically looking for it might be more robust than tracking the config. Also, the -emu interface doesn't appear until after boot and it is somewhat useful to be able to report that the domU has one or two or three interfaces even before it has booted, and we do that by reporting the existence of the non-emu interfaces.

**Comment by kalyan-nidumolu** [comment]
> Dynamically looking for it might be more robust than tracking the config. 

I am not sure of that. I prefer things to be driven by config - Gives us more control..

Any anyways - I guess till we see the final decision on config that would trigger emu, we can't come out with a config based knob? A few more Qs:

1) Today, Vifs are coming from NetworkInstance - So there are questions on . - Can same network instance be attached to both the types of apps

2) Can the same interface has both the -emu and non-emu versions of the interface - And when does this happen?

Anyways - if you think the approach is fine - go ahead. I don't know the full picture yet to suggest any alternate approaches..

**Pull request merged**